models/UserModel.js

```diff
@@ -14,7 +14,29 @@ const updateUserInfo = `
   RETURNING "userId", "displayName", "username";
 `;
 
+const getUsers = `
+  SELECT 
+      u."userId",
+      u."displayName",
+      u.username
+  FROM 
+      public.app_user u
+  LEFT JOIN 
+      public.follow f ON u."userId" = f."followedUserId"
```

Comment:
Is this join needed? I think it adds duplicate entries because we're adding another row for every user id in app_user that is also present in follow.followedUserId

![image](https://github.com/user-attachments/assets/fd52f660-e138-4c7c-99ee-8961c3abf579)

I think if the join gets removed we can also remove the group by clause on line 29. I suspect that group by clause is there to filter out the duplicate entries caused by this join

Reply:
I tried this out myself and I see the error it provides. I am going to change it so that it will look as such:

```
SELECT
    u."userId",
    u."displayName",
    u.username
FROM
    app_user u
WHERE
    u.username ILIKE '%' || $1 || '%'
    OR u."displayName" ILIKE '%' || $1 || '%'
ORDER BY
    (SELECT COUNT(f1."followedUserId")
     FROM follow f1
     WHERE f1."followedUserId" = u."userId") DESC
LIMIT 15;
```